metadata-ingestion/src/datahub/cli/ingest_cli.py

```diff
@@ -198,10 +198,12 @@ async def run_ingestion_and_check_upgrade() -> int:
         return ret
 
     loop = asyncio.get_event_loop()
-    ret = loop.run_until_complete(run_ingestion_and_check_upgrade())
-    if ret:
-        sys.exit(ret)
-    # don't raise SystemExit if there's no error
+    loop.run_until_complete(run_ingestion_and_check_upgrade())
+
+
+#     if ret:
```

Comment:
Can we keep this? 
I feel like this is a breaking change

Reply:
Keeping this causes the ingestion to stop if there are failures

Reply:
I have tested this in our production environment with 2.3 million datasets. And I didn't face any issues

Reply:
Any idea under what conditions it could fail?

Reply:
if there ingestion has failures, we want a non-zero return code. this makes that happen, and so it cannot be removed.

Reply:
Reverting the change for now as I don't have the logs where it was getting exited for certain failures.
Will do the testing again and report with the error message.